modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/edit_kaleo_definition_version.jsp

```diff
@@ -332,109 +332,72 @@ String successMessageKey = KaleoDesignerPortletKeys.KALEO_DESIGNER + "requestPro
+									function <portlet:namespace />afterTabViewChange(event) {
+										var tabContentNode = event.newVal.get('boundingBox');
 
-											var kaleoDesigner = <portlet:namespace />kaleoDesigner;
+										var kaleoDesigner = <portlet:namespace />kaleoDesigner;
 
-											if (tabContentNode === kaleoDesigner.viewNode && kaleoDesigner.editor) {
-												setTimeout(function () {
-													kaleoDesigner.set(
-														'definition',
-														kaleoDesigner.editor.get('value')
-													);
-												}, 0);
-											}
-										},
-										['aui-base']
-									);
-
-									Liferay.provide(
-										window,
-										'<portlet:namespace />publishKaleoDefinitionVersion',
-										function () {
-											<portlet:namespace />updateContent();
-
-											<portlet:namespace />updateTitle();
+										if (tabContentNode === kaleoDesigner.viewNode && kaleoDesigner.editor) {
+											setTimeout(function () {
+												kaleoDesigner.set('definition', kaleoDesigner.editor.get('value'));
+											}, 0);
+										}
+									}
 
-											<portlet:namespace />updateAction(
-												'<portlet:actionURL name="publishKaleoDefinitionVersion" />'
-											);
+									function <portlet:namespace />publishKaleoDefinitionVersion() {
+										<portlet:namespace />updateContent();
 
-											submitForm(document.<portlet:namespace />fm);
-										},
-										['aui-base']
-									);
+										<portlet:namespace />updateTitle();
 
-									Liferay.provide(
-										window,
-										'<portlet:namespace />saveKaleoDefinitionVersion',
-										function () {
-											<portlet:namespace />updateContent();
+										<portlet:namespace />updateAction(
+											'<portlet:actionURL name="publishKaleoDefinitionVersion" />'
+										);
 
-											<portlet:namespace />updateTitle();
+										submitForm(document.<portlet:namespace />fm);
+									}
 
-											<portlet:namespace />updateAction(
-												'<portlet:actionURL name="saveKaleoDefinitionVersion" />'
-											);
+									function <portlet:namespace />saveKaleoDefinitionVersion() {
+										<portlet:namespace />updateContent();
 
-											submitForm(document.<portlet:namespace />fm);
-										},
-										['aui-base']
-									);
+										<portlet:namespace />updateTitle();
 
-									Liferay.provide(
-										window,
-										'<portlet:namespace />updateAction',
-										function (action) {
-											var A = AUI();
+										<portlet:namespace />updateAction(
+											'<portlet:actionURL name="saveKaleoDefinitionVersion" />'
+										);
 
-											var form = A.one(document.<portlet:namespace />fm);
+										submitForm(document.<portlet:namespace />fm);
+									}
 
-											form.attr('action', action);
-										},
-										['aui-base']
-									);
+									function <portlet:namespace />updateAction(action) {
+										var form = document.<portlet:namespace />fm;
```

Comment:
I see we do this in many places, just caught my attention now... @wincent, is this something we'd like to do or stop doing? 

I was perusing to find this in the spec (I remember reading about it a long time ago) and couldn't find it quickly. Closer I got to this was the [`forms`](https://html.spec.whatwg.org/multipage/dom.html#dom-document-forms) element which is probably yet another way of doing this.

Just for the future, not for this... do we want to:
- `document.getElementById('<portlet:namespace />fm')`
- `document.<portlet:namespace />fm`
- `document.forms['<portlet:namespace />fm']`

Does it even matter?

Reply:
That's a good question, I forgot to mention it, but I was also surprised to see this.

Reply:
> Closer I got to this was the [`forms`](https://html.spec.whatwg.org/multipage/dom.html#dom-document-forms) element which is probably yet another way of doing this.

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/forms) says IE compatibility is unknown on that one, so that would be something to check.

It does seem a bit sneaky to go sticking this on the `document` like that, as if one global (`window`) weren't already room enough for us. Seems like the kind of thing that we could probably refactor away from without too much risk of breakage (who on earth would be depending on this), but neither do I see it as particularly high value (and obviously not something we need to tackle as part of this PR).

Reply:
> [...] (who on earth would be depending on this) [...]

You'd be surprised... or wouldn't you?

https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/form.js#L477

😂

Reply:
If it's just us, then that's fine, because we can change it.